File: frontend/src/services/dashboardService.js
import api from './api';

export const dashboardService = {
  getDashboard: async () => {
    return api.get('/dashboard');
  },
  getPrincipalDashboard: async () => {
    return api.get('/dashboard/principal');
  },
  getHodDashboard: async () => {
    return api.get('/dashboard/hod');
  },
  getTeacherDashboard: async () => {
    return api.get('/dashboard/teacher');
  },
  getStudentDashboard: async () => {
    return api.get('/dashboard/student');
  }
};

File: backend/src/main/java/com/university/sms/service/DashboardService.java
package com.university.sms.service;

import com.university.sms.dto.response.DashboardResponse;

import com.university.sms.security.UserPrincipal;

public interface DashboardService {
    DashboardResponse getStudentDashboard(Long userId);
    DashboardResponse getPrincipalDashboard(Long userId);
    DashboardResponse getHodDashboard(Long userId);
    DashboardResponse getTeacherDashboard(Long userId);
    DashboardResponse getDashboardByRole(UserPrincipal currentUser);
}
